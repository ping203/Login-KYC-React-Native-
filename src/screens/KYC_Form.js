import React, {Component} from 'react';
import {TextInput, View, Text, TouchableOpacity} from 'react-native';

import styles from '../components/styles';

export default class KYC_Form extends Component {
  constructor(props) {
    super(props);

    this.state = {
      TextInput_Name: '',
      TextInput_Gender: '',
      TextInput_Dob: '',
      TextInput_WorkExp: '',
    };
  }

  Send_Data_Function = () => {
    this.props.navigation.navigate('Profile', {
      NameOBJ: this.state.TextInput_Name,
      GenderOBJ: this.state.TextInput_Gender,
      DobOBJ: this.state.TextInput_Dob,
      WorkExpOBJ: this.state.TextInput_WorkExp,
    });
  };
  next = () => {
    this.props.navigation.navigate('KYC_Pic');
  };
  render() {
    return (
      <View style={styles.MainContainer}>
        <Text style={styles.kycId}>Please Enter following Details</Text>
        <TextInput
          placeholder="Name"
          onChangeText={data => this.setState({TextInput_Name: data})}
          style={styles.textInputStyle}
          underlineColorAndroid="transparent"
        />
        <TextInput
          placeholder="Gender"
          onChangeText={data => this.setState({TextInput_Gender: data})}
          style={styles.textInputStyle}
          underlineColorAndroid="transparent"
        />
        <TextInput
          placeholder="Date of Birth"
          onChangeText={data => this.setState({TextInput_Dob: data})}
          style={styles.textInputStyle}
          keyboardType={'numeric'}
          underlineColorAndroid="transparent"
        />
        <TextInput
          placeholder="Work Experience"
          onChangeText={data => this.setState({TextInput_WorkExp: data})}
          style={styles.textInputStyle}
          underlineColorAndroid="transparent"
        />

        <TouchableOpacity
          onPress={this.Send_Data_Function}
          activeOpacity={0.7}
          style={styles.button}>
          <Text style={styles.buttonText}> Save Details </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.KYCContainerStyle}
          activeOpacity={0.5}
          onPress={this.next}>
          <View style={styles.homeButton}>
            <Text style={styles.homeButtonStyle}>Next</Text>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}
