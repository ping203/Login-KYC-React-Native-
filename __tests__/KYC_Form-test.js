import React from 'react';
import KYC_Form from '../src/screens/KYC_Form';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<KYC_Form />).toJSON();
  expect(tree).toMatchSnapshot();
});
