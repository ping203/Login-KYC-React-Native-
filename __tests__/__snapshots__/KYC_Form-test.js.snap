// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Please Enter following Details
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Name"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#028b53",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 15,
        "textAlign": "center",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Gender"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#028b53",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 15,
        "textAlign": "center",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Date of Birth"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#028b53",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 15,
        "textAlign": "center",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Work Experience"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#028b53",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 15,
        "textAlign": "center",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00acee",
        "borderRadius": 8,
        "height": 40,
        "marginTop": 15,
        "opacity": 1,
        "padding": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
       Save Details 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 50,
        "marginTop": "10%",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00acee",
          "height": 50,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text>
        Next
      </Text>
    </View>
  </View>
</View>
`;
